lib/src/model/model_node.dart

```diff
@@ -67,6 +67,12 @@ class ModelNode {
         source = model_utils.stripIndentFromSource(source);
         source = model_utils.stripDartdocCommentsFromSource(source);
 
+        // The calculated source code end for field elements doesn't include
+        // the semicolon for field elements, so it has to be added manually
+        if (element is FieldElement) {
+          source += ';';
```

Comment:
This seems reasonable enough unless analyzer experts have a better idea of how to solve this.  I had thought it was just late fields that were broken, didn't realize it was all simple fields.  @scheglov ?

Reply:
This is mostly up to you. What `AstNode` do you expect to get for each documentable entity? For methods this is probably uncontroversial - `MethodDeclaration`. For fields - probably `VariableDeclaration`, with parent `VariableDeclarationList`, with parent `FieldDeclaration`. But you can have more than one field in one `FieldDeclaration`, so the whole idea of extracting "source code" for a field seems suspicious to me. Not knowing what exactly you are doing with it, I cannot say that it is definitely wrong though.